internal/kafka/internal/services/kafka.go

```diff
@@ -1044,23 +1057,9 @@ func (k *kafkaService) CountStreamingUnitByRegionAndInstanceType() ([]KafkaStrea
 		for i, streamingUnitCountPerRegion := range streamingUnitsCountPerRegion {
 			if streamingUnitCountPerRegion.isSame(kafkaCountPerRegion) {
 				streamingUnitsCountPerRegion[i].Count += streamingUnitCount
-				resultPresent = true
 				break
 			}
 		}
-
-		// We should never reach here in manual scaling
-		// TODO review if this is needed when dynamic scaling is implemented
-		if !resultPresent {
```

Comment:
This has been removed as a consideration in this work. 
For us to arrive here, it'll mean the kafka is not assigned to the cluster yet: 
- either, accepted but not assigned to a cluster 
- or accepted assigned to a cluster `A`, got rejected in the data plane, removed from cluster `A` and it'll be re-assigned onto another cluster at the next reconciler run


For all the above case, we'll have a different and dedicated metric for Kafkas that are not assigned to a cluster yet. 
This will be done as part of the work that's allows for a kafka to be accepted even when we do not have a ready data plane yet but regions limits have not been reached.